pkg/manager/internal.go

```diff
@@ -637,6 +638,10 @@ func (cm *controllerManager) startLeaderElection(ctx context.Context) (err error
 		return err
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Following @FillZpp's suggestion, I changed the location of initialization `cm.leaderElectionCancel` and `cm.leaderElectionStopped`.  So these values ​​will not be nil if `NewLeaderElector` succeeds.

 When `engageStopProcedure` needs to check `cm.leaderElectionCancel` and `cm.leaderElectionStopped`, it will first check whether they are nil.

Reply:
I'm not sure if this is correct, if we set the context and cancel here it'll be set in a goroutine which can cause a race condition if the controller closes too quickly.